kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/SharedInformerFactory.java

```diff
@@ -266,4 +385,12 @@ private void initOperationContext(Config configuration) {
       context = context.withIsNamespaceConfiguredFromGlobalConfig(true);
     }
   }
+
+  private <T extends HasMetadata> void  registerKindToKubernetesDeserializer(OperationContext context, Class<T> apiTypeClass) {
+    if (context.getApiGroupName() != null && context.getApiGroupVersion() != null) {
```

Comment:
Shouldn't this use the new `HasMetadata#getApiVersion` instead? Is there a scenario where the the `apiVersion` derived from the `OperationContext` would clash with the version provided by the class itself?